shuup/simple_supplier/utils.py

```diff
@@ -36,23 +37,26 @@ def get_current_stock_value(supplier_id, product_id):
     events = (
         StockAdjustment.objects
         .filter(supplier_id=supplier_id, product_id=product_id)
+        .exclude(type=StockAdjustmentType.RESTOCK_LOGICAL)
         .aggregate(total=Sum("delta"))["total"] or 0)
     orders_bought = (
         OrderLine.objects
         .filter(supplier_id=supplier_id, product_id=product_id)
         .exclude(
             Q(order__status__role=OrderStatusRole.CANCELED) |
-            Q(type=OrderLineType.AMOUNT_REFUND) |
-            Q(type=OrderLineType.QUANTITY_REFUND)
-        )
+            Q(type=OrderLineType.REFUND))
         .aggregate(total=Sum("quantity"))["total"] or 0)
+    orders_refunded_before_shipment = (
+        StockAdjustment.objects
+        .filter(supplier_id=supplier_id, product_id=product_id, type=StockAdjustmentType.RESTOCK_LOGICAL)
+        .aggregate(total=Sum("delta"))["total"] or 0)
     orders_sent = (
         ShipmentProduct.objects
         .filter(shipment__supplier=supplier_id, product_id=product_id)
         .aggregate(total=Sum("quantity"))["total"] or 0)
 
     return {
-        "logical_count": events - orders_bought,
+        "logical_count": events - orders_bought + orders_refunded_before_shipment,
         "physical_count": events - orders_sent
     }
```

Comment:
Front should be using `StockCount`/`get_stock_status` to get cached stock counts. 

Agreed that this function is db intensive but updating this (and likely the supplier API) is out-of-scope IMO.